pkg/services/mysql/common_provision.go

```diff
@@ -13,7 +13,28 @@ import (
 	uuid "github.com/satori/go.uuid"
 )
 
-func validateDBMSProvisionParameters(pp dbmsProvisioningParameters) error {
+const (
+	enabled           = "enabled"
+	disabled          = "disabled"
+	enabledARMString  = "Enabled"
+	disabledARMString = "Disabled"
+)
+
+func validateDBMSProvisionParameters(
+	plan service.Plan,
+	pp dbmsProvisioningParameters,
+) error {
+	if plan == nil {
+		return fmt.Errorf("plan invalid")
+	}
+	s, ok := plan.GetProperties().Extended["provisionSchema"]
+	if !ok {
+		return fmt.Errorf("invalid plan, schema not found")
+	}
+	schema := s.(planSchema)
+	if err := schema.validateProvisionParameters(pp); err != nil {
+		return err
+	}
```

Comment:
I'm a little concerned with how fragmented this feels. It seems we're delegating most of the validation to this schema object that enumerates permissible values for a host of _new_ provisioning parameters, but we're still handing validation for all the previously existing provisioning parameters over the next several lines here. Not sure why we wouldn't do all of this in one place using a single approach.